src/useForm.tsx

```diff
@@ -144,6 +137,34 @@ function forEach<T>(
   }
 }
 
-function isEmptyArray(obj: any): boolean {
-  return Array.isArray(obj) && obj.length === 0
+/** Recursively visits each property in object
+ *  and returns true as soon as any field matches the passed
+ *  predicate, false otherwise.
+ */
+function exists<T>(
```

Comment:
So, previously we'd visit every property on our form's state (recursively) and only return at the end. Now we bail out early as soon as we find a property that matches our predicate (e.g. `!isEmptyObject`)

Reply:
Also yes yes this would technically be faster as a loop with a queue/stack (pick your poison BFS vs DFS) vs recursion -- but I'm pretty sure it's never going to matter ;).